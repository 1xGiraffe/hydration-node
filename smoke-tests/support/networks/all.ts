import { Config } from './types'

import hydraDXConfig from './hydraDX'
import polkadotConfig from './polkadot'

const all = {
  hydraDX: hydraDXConfig,
  polkadot: polkadotConfig,
} satisfies Record<string, Config>

export default all
