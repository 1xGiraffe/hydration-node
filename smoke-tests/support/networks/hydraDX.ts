import { Config } from './types'

export default {
  polkadot: {
    name: 'hydraDX' as const,
    endpoint: 'wss://rpc.hydradx.cloud',
  },
  kusama: {
    name: 'basilisk' as const,
    endpoint: 'wss://rpc.basilisk.cloud',
  },
  config: ({ alice }) => ({
    storages: {
      System: {
        Account: [[[alice.address], {data: {free: "1000000000000000000"}}]],
      },
      Tokens: {
        accounts: [
          [[alice.address, 2], {free: '0x0000000000000000000000000000000000000000000000056bc75e2d6310000'}],
          [[alice.address, 5], {free: 100e10}],
        ],
      },
      Council: {
        members: [
          alice.address,
        ]
      },
      TechnicalCommittee: {
        members: [
          alice.address,
        ]
      },


    },
  }),
} satisfies Config

export const hydraDX = {
  paraId: 2034,
  dai: 2,
  hdx: 0,
  dot: 5,
}

export const basilisk = {
  paraId: 2090,
  dai: 13,
}
