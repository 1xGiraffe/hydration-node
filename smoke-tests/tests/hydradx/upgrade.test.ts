import { afterAll, assert, beforeEach, describe, it } from 'vitest'
import { sendTransaction, testingPairs } from '@acala-network/chopsticks-testing'

import { checkEvents } from '../../support/helpers'
import { createNetworks } from '../../support/networks'

import * as fs from 'fs'

import { blake2AsHex } from "@polkadot/util-crypto";
describe.each([
  {
    name: 'hydraDX',
  },
] as const)('$name upgrade', async ({ name}) => {
  const { [name]: chain } = await createNetworks({ [name]: undefined })
  const { alice } = testingPairs()

  const head = chain.chain.head

  const code = fs.readFileSync('./38.wasm').toString()

  afterAll(async () => {
    await chain.teardown()
  })

  beforeEach(async () => {
    await chain.chain.setHead(head)
  })

  it.each([
    {
      name: 'gov upgrade',
    }
  ])('$name works', async () => {
    const proposal = chain.api.tx.parachainSystem.authorizeUpgrade(blake2AsHex(code))
    const encodedProposal = proposal.method.toHex()
    console.log(encodedProposal)
    let encodedHash = blake2AsHex(encodedProposal);

    const tx11 = chain.api.tx.preimage
      .notePreimage(encodedProposal)

    const tx0 = await sendTransaction(tx11.signAsync(alice))
    await chain.chain.newBlock()
    await checkEvents(tx0, "preimage").redact({ number: true }).toMatchSnapshot()

    let external = chain.api.tx.democracy.externalProposeMajority(encodedHash);
    let fastTrack = chain.api.tx.democracy.fastTrack(encodedHash, 2, 1);
    let referendumNextIndex = (await chain.api.query.democracy.referendumCount()).toNumber();
    console.log(referendumNextIndex)

    const voteAmount = 1n * 10n ** BigInt(chain.api.registry.chainDecimals[0]);

    console.log("council")
    const c1 = chain.api.tx.council
      .propose(1, external, external.length)
    const c1tx = await sendTransaction(c1.signAsync(alice))
    await chain.chain.newBlock()
    await checkEvents(c1tx).redact({ number: true }).toMatchSnapshot()

    console.log("tch committee")
    const tc = chain.api.tx.technicalCommittee
      .propose(1, fastTrack, fastTrack.length)
    const tctx = await sendTransaction(tc.signAsync(alice))
    await chain.chain.newBlock()
    await checkEvents(tctx).redact({ number: true }).toMatchSnapshot("tech committee")


    console.log("vote")
    const voteTx = chain.api.tx.democracy
      .vote(referendumNextIndex, {
        Standard: {
          balance: voteAmount,
          vote: { aye: true, conviction: 1 },
        },
      });

    const voteTx2 = await sendTransaction(voteTx.signAsync(alice))
    await chain.chain.newBlock()
    await checkEvents(voteTx2).redact({ number: true }).toMatchSnapshot("vote")

    let entries = await chain.api.query.democracy.referendumInfoOf.entries()

    for (let entry of entries) {
      let idx = chain.api.registry.createType("u32", entry[0].toU8a().slice(-4)).toNumber()
      if (idx == referendumNextIndex) {
        console.log(idx)
        let f = entry[1].unwrap().isFinished;
        console.log(f)

      }
    }

    console.log("enact")
    const enact = chain.api.tx.parachainSystem.enactAuthorizedUpgrade(code)
    const enactTx = await sendTransaction(enact.signAsync(alice))
    await chain.chain.newBlock()
    await checkEvents(enactTx).toMatchSnapshot("enact")

    console.log(chain.api.runtimeVersion.specVersion.toNumber())

  })
})

