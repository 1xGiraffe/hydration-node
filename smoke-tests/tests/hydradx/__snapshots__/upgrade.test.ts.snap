// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`'hydraDX' upgrade > 'gov upgrade' works > enact 1`] = `[]`;

exports[`'hydraDX' upgrade > 'gov upgrade' works > events 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
    },
    "method": "Noted",
    "section": "preimage",
  },
]
`;

exports[`'hydraDX' upgrade > 'gov upgrade' works > events 2`] = `
[
  {
    "data": {
      "amount": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "Withdraw",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(number)",
      "currencyId": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "Withdrawn",
    "section": "currencies",
  },
  {
    "data": {
      "proposalHash": "(hash)",
      "result": "Ok",
    },
    "method": "Executed",
    "section": "council",
  },
  {
    "data": {
      "amount": "(number)",
      "who": "7L53bUTBopuwFt3mKUfmkzgGLayYa1Yvn1hAg9v5UMrQzTfh",
    },
    "method": "Deposit",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(number)",
      "currencyId": "(number)",
      "who": "7L53bUTBopuwFt3mKUfmkzgGLayYa1Yvn1hAg9v5UMrQzTfh",
    },
    "method": "Deposited",
    "section": "currencies",
  },
  {
    "data": {
      "actualFee": "(number)",
      "tip": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "TransactionFeePaid",
    "section": "transactionPayment",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Operational",
        "paysFee": "Yes",
        "weight": "(number)",
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`'hydraDX' upgrade > 'gov upgrade' works > tech committee 1`] = `
[
  {
    "data": {
      "amount": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "Withdraw",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(number)",
      "currencyId": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "Withdrawn",
    "section": "currencies",
  },
  {
    "data": {
      "refIndex": "(number)",
      "threshold": "SimpleMajority",
    },
    "method": "Started",
    "section": "democracy",
  },
  {
    "data": {
      "proposalHash": "(hash)",
      "result": "Ok",
    },
    "method": "Executed",
    "section": "technicalCommittee",
  },
  {
    "data": {
      "amount": "(number)",
      "who": "7L53bUTBopuwFt3mKUfmkzgGLayYa1Yvn1hAg9v5UMrQzTfh",
    },
    "method": "Deposit",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(number)",
      "currencyId": "(number)",
      "who": "7L53bUTBopuwFt3mKUfmkzgGLayYa1Yvn1hAg9v5UMrQzTfh",
    },
    "method": "Deposited",
    "section": "currencies",
  },
  {
    "data": {
      "actualFee": "(number)",
      "tip": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "TransactionFeePaid",
    "section": "transactionPayment",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Operational",
        "paysFee": "Yes",
        "weight": "(number)",
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`'hydraDX' upgrade > 'gov upgrade' works > vote 1`] = `
[
  {
    "data": {
      "amount": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "Withdraw",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(number)",
      "currencyId": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "Withdrawn",
    "section": "currencies",
  },
  {
    "data": {
      "refIndex": "(number)",
      "vote": {
        "Standard": {
          "balance": "(number)",
          "vote": {
            "conviction": "Locked1x",
            "vote": "Aye",
          },
        },
      },
      "voter": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "Voted",
    "section": "democracy",
  },
  {
    "data": {
      "amount": "(number)",
      "who": "7L53bUTBopuwFt3mKUfmkzgGLayYa1Yvn1hAg9v5UMrQzTfh",
    },
    "method": "Deposit",
    "section": "balances",
  },
  {
    "data": {
      "amount": "(number)",
      "currencyId": "(number)",
      "who": "7L53bUTBopuwFt3mKUfmkzgGLayYa1Yvn1hAg9v5UMrQzTfh",
    },
    "method": "Deposited",
    "section": "currencies",
  },
  {
    "data": {
      "actualFee": "(number)",
      "tip": "(number)",
      "who": "7Lh1DEaxk8TNVU7snC3UkSv7N7NHb7BUuEkYZBwrzfrPNQE3",
    },
    "method": "TransactionFeePaid",
    "section": "transactionPayment",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": "(number)",
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;
